baseplate/server/runtime_monitor.py

```diff
@@ -217,7 +218,7 @@ def _report_runtime_metrics_periodically(
 
 def start(server_config: Dict[str, str], application: Any, pool: Pool) -> None:
     baseplate: Baseplate = getattr(application, "baseplate", None)
-    if not baseplate or not baseplate._metrics_client:
+    if not baseplate or not prometheus_metrics.is_metrics_enabled(server_config):
         logger.info("No metrics client configured. Server metrics will not be sent.")
         return
```

Comment:
This check isn't really checking if it _should_ be on, but rather if it was instantiated properly. Prometheus also doesn't have a "metrics client" so the naming of this now seems weird.

Maybe something like:
```suggestion
    if not baseplate or not baseplate._metrics_client:
        logger.info("No statsd metrics configured, statsd metrics will not be sent. If you're expecting them, check configs for 'metrics.tagging' or 'metrics.namespace'")

    if not prometheus_metrics.is_metrics_enabled(server_config):
        logger.info("No metrics frameworks are enabled, skipping reporter setup"
        return
```

Reply:
If baseplate is None, which I'm not clear on when this would happen, accessing `baseplate._metrics_client` will fail. I could check for that case lower down.

Reply:
You could also do something like
```suggestion
    statsd_metrics_client = None
    if not baseplate or not baseplate._metrics_client:
        logger.info("No metrics client configured. Server metrics will not be sent.")
        // initialize noop client
    else:
        statsd_metrics_client = baseplate._metrics_client
```

Reply:
This matches to old code, I believe it's using `baseplate._metrics_client` as a gauge to see if metrics are enabled. This is no longer a valid way to determine if metrics are enabled when we use Prometheus.